## Table des matières

- [Question 1](#question1)
    - [Dataframe](#dataframe)
    - [Population mondiale en 2013](#popmondiale2013)
- [Question 2](#question2)
- [Question 3](#question3)
- [Question 4](#question4)
- [Question 5](#question5)
    - [Liste des 20 aliments les plus caloriques selon les ratios moyens](#reponse5_1)
    - [Liste de 5 aliments parmi les 20 les plus caloriques](#reponse5_2)
    - [Liste des 20 aliments les plus riches selon les pourcentages moyens](#reponse5_3)
    - [Liste de 5 aliments parmi les 20 les plus riches](#reponse5_4)    
- [Question 6](#question6)
- [Question 7](#question7)
- [Question 8](#question8)
- [Question 9](#question9)
- [Question 10](#question10)
- [Question 11](#question11)
- [Question 12](#question12)
    - [Informations, parmi les données des bilans alimentaires, relatives aux pays dans lesquels la FAO recence des personnes en sous-nutrition](#réponse12_1)
    - [Les 15 produits les plus exportés par les pays recensant des personnes en sous-nutrition](#réponse12_2)
    - [Classement des 200 plus grandes importations de ces 15 produits](#réponse12_3)
    - [Somme des 200 importations les plus importantes des 15 produits les plus exportés](#réponse12_4)
    - [Tableau pour chaque produit des ratios](#réponse12_5)
    - [Affichage des trois produits avec le ratio "Autres utilisations" / Disponibilité intérieure le plus élevé](#réponse12_6)
    - [Affichage des trois produits avec le ratio Aliments pour animaux / Nourriture totale (animale + humaine) le plus élevé](#réponse12_7)
- [Question13](#question13)
- [Question14](#question14)

In [1]:
import numpy as np
import pandas as pd

animaux = pd.read_csv("fr_animaux.csv")
cereales = pd.read_csv("fr_céréales.csv")
population = pd.read_csv("fr_population.csv")
sousalimentation = pd.read_csv("fr_sousalimentation.csv")
vegetaux = pd.read_csv("fr_vegetaux.csv")

<a id="question1"></a>
#### <span style="background-color: #FFFF00">Question 1 : créer un DataFrame affichant la population par pays puis calculer la population mondiale en 2013.</span>

In [2]:
population.head(3)

Code Domaine                                            Domaine  Code zone  \
0         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
1         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        202   
2         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          3   

             Zone  Code Élément            Élément  Code Produit     Produit  \
0     Afghanistan           511  Population totale          2501  Population   
1  Afrique du Sud           511  Population totale          2501  Population   
2         Albanie           511  Population totale          2501  Population   

   Code année  Année           Unité  Valeur Symbole Description du Symbole  
0        2013   2013  1000 personnes   30552     NaN      Donnée officielle  
1        2013   2013  1000 personnes   52776     NaN      Donnée officielle  
2        2013   2013  1000 personnes    3173     NaN      Donnée officielle

<a id="dataframe"></a>
#### DataFrame affichant la population par pays en 2013 :

In [3]:
# Calcul de la population mondiale

PopTotale2013 =  population.Valeur.sum()*1000
#8413993000

# La population mondiale en 2013 est de 7,083 milliards de personnes.
# Source : http://economiedurable.over-blog.com/article-la-population-mondiale-au-1er-janvier-2013-99484009.html
# Il y a donc des corrections à apporter.

Pop = population.drop_duplicates()

# Aucune colonne ne comporte de doublons.
# Mais la Chine apparaît deux fois :
# - Chine, continentale (ligne 36, code zone = 41), donnée officielle
# - Chine (ligne 33, code zone = 351), agrégat incluant des données non officielles
# Il faut donc supprimer la ligne 33.¶

# Suppression de l'élément "Chine" avec le code zone 

Pop = population[population["Code zone"]!=351]

# Projection pour ne conserver que les colonnes "Zone" et "Valeur"
# Renommage de la colonne 'Valeur'

Pop = Pop.loc[ : , ('Zone','Valeur')]
Pop = Pop.rename(columns={"Valeur":"Population en milliers d'habitants"})

Pop.head(3)

Zone  Population en milliers d'habitants
0     Afghanistan                               30552
1  Afrique du Sud                               52776
2         Albanie                                3173

<a id="popmondiale2013"></a>
#### Calcul de la population mondiale en 2013 :

In [4]:
print("En 2013, la population mondiale est de" , Pop["Population en milliers d'habitants"].sum()*1000, "personnes.")

En 2013, la population mondiale est de 6997326000 personnes.


#### Avant d'aller plus loin : suppression de la zone 'Chine' dans les autres tableaux

In [5]:
animaux = animaux[animaux["Code zone"]!=351]
cereales = cereales[cereales["Code zone"]!=351]
sousalimentation = sousalimentation[sousalimentation["Code zone"]!=351]
vegetaux = vegetaux[vegetaux["Code zone"]!=351]

<a id="question2"></a>
#### <span style="background-color: #FFFF00">Question 2 : Identifier les redondances parmi les 11 informations (voir énoncé) et donner une réponse sous la forme d'une équation à trois termes faisant intervenir ces 11 quantités.</span>

#### 	disponibilité intérieure =
- disponibilité alimentaire (en milliers de tonnes) + aliments pour animaux + semences + pertes + traitements + autres utilisations
- production + importations - exportations + variations des stocks

#### Vérification avec l'exemple du blé en France :

In [6]:
vegetaux.head(3)

Code Domaine                                            Domaine  Code zone  \
0         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
1         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
2         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   

          Zone  Code Élément                  Élément  Code Produit Produit  \
0  Afghanistan          5511               Production          2511     Blé   
1  Afghanistan          5611  Importations - Quantité          2511     Blé   
2  Afghanistan          5072       Variation de stock          2511     Blé   

   Code année  Année               Unité  Valeur Symbole  \
0        2013   2013  Milliers de tonnes  5169.0       S   
1        2013   2013  Milliers de tonnes  1173.0       S   
2        2013   2013  Milliers de tonnes  -350.0       S   

  Description du Symbole  
0  Données standardisées  
1  Données standardisées  
2  Données standardisées

In [7]:
# Création d'un tableau affichant toutes les informations liées au blé en France

Tableau2 = vegetaux[vegetaux.Zone=='France']
Tableau2 = Tableau2[Tableau2.Produit=='Blé']
Tableau2 = Tableau2.pivot(index ='Produit', columns='Élément', values='Valeur')

# Calcul de la population française

PopFrance = Pop[Pop.Zone == "France"]
PopFrance = PopFrance["Population en milliers d'habitants"].sum()*1000

# Création du tableau avec une colonne par égalité

Tableau2['disponibilité alimentaire (en milliers de tonnes) + aliments pour animaux + semences + pertes + traitements + autres utilisations'] = (Tableau2['Disponibilité alimentaire en quantité (kg/personne/an)']*PopFrance/1000000)+Tableau2['Aliments pour animaux']+Tableau2['Semences']+Tableau2['Pertes']+Tableau2['Traitement']+Tableau2['Autres utilisations (non alimentaire)']
Tableau2['production + importations - exportations + variations des stocks'] = Tableau2['Production']+Tableau2['Importations - Quantité']-Tableau2['Exportations - Quantité']+Tableau2['Variation de stock']
Tableau2= Tableau2.loc[ : ,('Disponibilité intérieure','disponibilité alimentaire (en milliers de tonnes) + aliments pour animaux + semences + pertes + traitements + autres utilisations','production + importations - exportations + variations des stocks')]

Tableau2

# Les valeurs sont égales

Élément  Disponibilité intérieure  \
Produit                             
Blé                       20298.0   

Élément  disponibilité alimentaire (en milliers de tonnes) + aliments pour animaux + semences + pertes + traitements + autres utilisations  \
Produit                                                                                                                                      
Blé                                            20298.07313                                                                                   

Élément  production + importations - exportations + variations des stocks  
Produit                                                                    
Blé                                                20298.0

<a id="question3"></a>
#### <span style="background-color: #FFFF00">Question 3 : Calcul (pour chaque pays et chaque produit) de la disponibilité alimentaire en kcal puis en kg par protéines</span>

In [8]:
data = pd.concat([animaux,vegetaux,cereales])
data= pd.merge(Pop,data)

data.head(3)

Zone  Population en milliers d'habitants Code Domaine  \
0  Afghanistan                               30552         FBSH   
1  Afghanistan                               30552         FBSH   
2  Afghanistan                               30552         FBSH   

                                             Domaine  Code zone  Code Élément  \
0  Bilans Alimentaire (Ancienne méthodologie et p...          2          5511   
1  Bilans Alimentaire (Ancienne méthodologie et p...          2          5611   
2  Bilans Alimentaire (Ancienne méthodologie et p...          2          5301   

                    Élément  Code Produit           Produit  Code année  \
0                Production          2731  Viande de Bovins        2013   
1   Importations - Quantité          2731  Viande de Bovins        2013   
2  Disponibilité intérieure          2731  Viande de Bovins        2013   

   Année               Unité  Valeur Symbole Description du Symbole  
0   2013  Milliers de tonnes   134.0       S  Données standardisées  
1   2013  Milliers de tonnes     6.0       S  Données standardisées  
2   2013  Milliers de tonnes   140.0       S  Données standardisées

In [9]:
# Création du tableau 3

Tableau3 = data.pivot_table(index =['Zone','Code zone',"Population en milliers d'habitants",'Produit'],
                            columns='Élément', values='Valeur', aggfunc = sum)
Tableau3 = Tableau3.loc[ : ,('Disponibilité alimentaire (Kcal/personne/jour)','Disponibilité de protéines en quantité (g/personne/jour)')]
Tableau3 = Tableau3.reset_index()

# Conversion puis renommage des colonnes

Tableau3['Disponibilité de protéines en quantité (g/personne/jour)'] = (Tableau3['Disponibilité de protéines en quantité (g/personne/jour)']*Tableau3["Population en milliers d'habitants"]*1000*365)/1000
Tableau3['Disponibilité alimentaire (Kcal/personne/jour)'] = Tableau3['Disponibilité alimentaire (Kcal/personne/jour)']*Tableau3["Population en milliers d'habitants"]*1000*365

Tableau3= Tableau3.rename(columns={'Disponibilité alimentaire (Kcal/personne/jour)':'Disponibilité alimentaire (Kcal/an)', 'Disponibilité de protéines en quantité (g/personne/jour)':'Disponibilité de protéines en quantité (kg/an)'})

# Suppression des valeurs incohérentes

Tableau3 = Tableau3.drop(Tableau3[Tableau3['Disponibilité alimentaire (Kcal/an)']<0].index)
Tableau3 = Tableau3.drop(Tableau3[Tableau3['Disponibilité de protéines en quantité (kg/an)']<0].index)
Tableau3 = Tableau3.drop_duplicates()

Tableau3.head(3)

Élément         Zone  Code zone  Population en milliers d'habitants  \
0        Afghanistan          2                               30552   
1        Afghanistan          2                               30552   
2        Afghanistan          2                               30552   

Élément                Produit  Disponibilité alimentaire (Kcal/an)  \
0             Abats Comestible                         5.575740e+10   
1              Agrumes, Autres                         1.115148e+10   
2        Aliments pour enfants                         1.115148e+10   

Élément  Disponibilité de protéines en quantité (kg/an)  
0                                             8586639.6  
1                                              223029.6  
2                                              334544.4

<a id="question4"></a>
#### <span style="background-color: #FFFF00">Question 4 : calculer pour chaque produit le ratio "énergie/poids" en kcal/kg. Vérifier la cohérence de ce résultat en comparant ce ratio aux données disponibles sur Internet, avec par exemple la valeur calorique d'un oeuf. Calculer également le pourcentage de protéines de chaque produit pour chaque pays. Vérifier la cohérence du calcul en comparant les résultats avec la teneur en protéines de l'avoine.</span>

In [10]:
Tableau4 = data.pivot_table(index =['Zone','Code zone',"Population en milliers d'habitants",'Produit'],
                            columns='Élément', values='Valeur', aggfunc = sum)
Tableau4 = Tableau4.loc[ : ,('Disponibilité alimentaire en quantité (kg/personne/an)')]
Tableau4 = Tableau4.reset_index()
Tableau4 = pd.merge (Tableau4,Tableau3)

# Suppression des valeurs incohérentes

Tableau4 = Tableau4.drop(Tableau4[Tableau4['Disponibilité alimentaire en quantité (kg/personne/an)']<0].index)
Tableau4 = Tableau4.drop_duplicates()

# Création des nouvelles colonnes

Tableau4['Disponibilité alimentaire en quantité (kg/personne/an)'] = Tableau4['Disponibilité alimentaire en quantité (kg/personne/an)']*Tableau4["Population en milliers d'habitants"]*1000
Tableau4['Ratio énergie/poids en kcal/kg'] = Tableau4['Disponibilité alimentaire (Kcal/an)'] / Tableau4['Disponibilité alimentaire en quantité (kg/personne/an)']
Tableau4['Pourcentage de protéines'] = (Tableau4['Disponibilité de protéines en quantité (kg/an)'] / Tableau4['Disponibilité alimentaire en quantité (kg/personne/an)'])*100
Tableau4 = Tableau4.loc[ : ,('Zone','Produit','Ratio énergie/poids en kcal/kg','Pourcentage de protéines')]

Tableau4.head(3)

Zone                Produit  Ratio énergie/poids en kcal/kg  \
0  Afghanistan       Abats Comestible                     1061.046512   
1  Afghanistan        Agrumes, Autres                      282.945736   
2  Afghanistan  Aliments pour enfants                     6083.333333   

   Pourcentage de protéines  
0                 16.340116  
1                  0.565891  
2                 18.250000

####  Comparaison des résultats de ratio avec celui d'un oeuf :
- La valeur calorique d'un oeuf est de 147 kcal pour 100 g soit un ratio énergie/poids de 1470 kcal/kg.
- Source : https://fr.wikipedia.org/wiki/%C5%92uf_(aliment)#Valeur_nutritionnelle

In [11]:
# Pour comparer mes résultats au ratio d'un oeuf et vérifier leur cohérence,
# j'affiche les statistiques de Tableau4

Tableau4.describe(include="all")

Zone        Produit  Ratio énergie/poids en kcal/kg  \
count   15604          15604                    1.208500e+04   
unique    174             98                             NaN   
top      Inde  Viande, Autre                             NaN   
freq       97            174                             NaN   
mean      NaN            NaN                             inf   
std       NaN            NaN                             NaN   
min       NaN            NaN                    0.000000e+00   
25%       NaN            NaN                    3.313162e+02   
50%       NaN            NaN                    1.183784e+03   
75%       NaN            NaN                    3.421875e+03   
max       NaN            NaN                             inf   

        Pourcentage de protéines  
count               10063.000000  
unique                       NaN  
top                          NaN  
freq                         NaN  
mean                    7.055759  
std                     7.475088  
min                     0.000000  
25%                     0.604542  
50%                     5.639263  
75%                    11.398117  
max                    42.583333

In [12]:
# La moyenne et le maximum sont à l'infini
# Suppression des valeurs nulles ou Nan

Tableau4 = Tableau4.dropna()

T4_Stat = Tableau4.describe(include="all")

T4_Stat["Ratio énergie/poids en kcal/kg"]

count     10063.000000
unique             NaN
top                NaN
freq               NaN
mean       1926.004271
std        2286.494649
min           0.000000
25%         319.505522
50%         977.678571
75%        2963.367250
max       21900.000000
Name: Ratio énergie/poids en kcal/kg, dtype: float64

#### La moyenne du ratio de 1926 kcal/kg est du même ordre que la valeur du ratio de l'oeuf. La valeur maximale de 21900 kcal/kg est incohérente.
#### Recherche du produit correspondant et comparaison avec sa valeur sur Internet :

In [13]:
# Recherche de l'index de la ligne avec la valeur maximale de ratio: 

Tableau4["Ratio énergie/poids en kcal/kg"].argmax()

1894

In [14]:
# Affichage des informations de la ligne 1894

Tableau4.iloc[1894]

Zone                                                Chili
Produit                           Huil Plantes Oleif Autr
Ratio énergie/poids en kcal/kg                      21900
Pourcentage de protéines                             21.9
Name: 2931, dtype: object

#### C'est l'huile de moringa oleifera du Chili qui affiche un ratio de 21900 kcal/kg. Or, le ratio sur Internet donne 370 kcal/kg.
Source : https://fr.wikipedia.org/wiki/Moringa_oleifera#Huile_des_graines
#### Ce résultat est incohérent.

#### Comparaison de l'huile de moringa oleifera du Chili avec celle des autres pays :

In [15]:
T4Oleif = Tableau4[Tableau4["Produit"]== "Huil Plantes Oleif Autr"]

T4Oleif.sort_values(by="Ratio énergie/poids en kcal/kg").tail(3)

Zone                  Produit  Ratio énergie/poids en kcal/kg  \
4405    Espagne  Huil Plantes Oleif Autr                         18250.0   
5510  Guatemala  Huil Plantes Oleif Autr                         18250.0   
2931      Chili  Huil Plantes Oleif Autr                         21900.0   

      Pourcentage de protéines  
4405                       0.0  
5510                       0.0  
2931                      21.9

In [16]:
T4Oleif.describe()

Ratio énergie/poids en kcal/kg  Pourcentage de protéines
count                      133.000000                133.000000
mean                      8425.052890                  0.571416
std                       2501.800924                  1.954110
min                          0.000000                  0.000000
25%                       8111.111111                  0.000000
50%                       8620.603015                  0.345154
75%                       8810.344828                  0.670956
max                      21900.000000                 21.900000

#### La moyenne des huiles de moringa oleifera est de 8425 kcal/kg.
Cette moyenne est beaucoup plus basse que le ratio de l'huile du Chili. Cependant, l'huile du Guatemala et d'Espagne ont elles aussi un ratio très élevé (18250 kcal/kg). Le ratio de l'huile de moringa oleifera du Chili paraît donc cohérent par comparaison aux autres huiles de moringa oleifera.

####  Comparaison des résultats du pourcentage de protéines avec celui de l'avoine :
La composition en protéines de 100 g d'avoine est de 10,7 g, soit un pourcentage de protéines de 10,7%.
#### Source : https://fr.wikipedia.org/wiki/Avoine_cultiv%C3%A9e

In [17]:
T4_Stat["Pourcentage de protéines"]

count     10063.000000
unique             NaN
top                NaN
freq               NaN
mean          7.055759
std           7.475088
min           0.000000
25%           0.604542
50%           5.639263
75%          11.398117
max          42.583333
Name: Pourcentage de protéines, dtype: float64

#### La moyenne des pourcentages de protéines est de 7% et la moyenne de 75% des valeurs est de 11,4%.
Ces résultats sont cohérents par comparaison au pourcentage de l'avoine.
Cependant le maximum de 42,58% est bien supérieur.

#### Identification du produit correspondant:

In [18]:
Tableau4["Pourcentage de protéines"].argmax()

5973

In [19]:
Tableau4.iloc[5973]

Zone                                         Mongolie
Produit                           Poissons Pelagiques
Ratio énergie/poids en kcal/kg                6083.33
Pourcentage de protéines                      42.5833
Name: 9220, dtype: object

#### Il s'agit des poissons pélagiques de Mongolie.
Comparaison avec les poissons pélagiques des autres pays.

In [20]:
T4PoissonPelagiques = Tableau4[Tableau4["Produit"]== "Poissons Pelagiques"]

T4PoissonPelagiques.sort_values(by="Pourcentage de protéines").tail(3)

Zone              Produit  Ratio énergie/poids en kcal/kg  \
7094      Japon  Poissons Pelagiques                     1426.436782   
11057   Pologne  Poissons Pelagiques                     2009.174312   
9220   Mongolie  Poissons Pelagiques                     6083.333333   

       Pourcentage de protéines  
7094                  20.081992  
11057                 27.709862  
9220                  42.583333

In [21]:
T4PoissonPelagiques.describe()

Ratio énergie/poids en kcal/kg  Pourcentage de protéines
count                      172.000000                172.000000
mean                       888.304389                 12.129966
std                        513.050752                  3.420235
min                          0.000000                  0.000000
25%                        806.630745                 10.848121
50%                        885.054839                 11.608869
75%                        973.395311                 12.466360
max                       6083.333333                 42.583333

#### Le pourcentage de protéines des poissons pélagiques de Mongolie est bien supérieur à celui des poissons pélagiques des autres pays. Cependant, le terme "pélagique" renvoie à la haute mer et ne définit donc pas une famille de poissons en particulier.
#### Les résultats sont donc considérés comme cohérents.
#### Source supplémentaire : http://www.fao.org/3/y5740f/y5740f0f.htm

<a id="question5"></a>
#### <span style="background-color: #FFFF00">Question 5 : Citer 5 aliments parmi les 20 aliments les plus caloriques en utilisant le ratio énergie/poids. Ce ratio peut varier selon le pays. Il faut donc d'abord réaliser une moyenne de chaque ratio de produit sur les pays. Créer une nouvelle table avec cette agrégation. Penser à retirer les valeurs nulles pour ne pas fausser le calcul de la moyenne. Citer 5 aliments parmi les plus riches en protéines.</span>

<a id="reponse5_1"></a>
#### Liste des 20 aliments les plus caloriques :

In [22]:
# Renommage de la colonne "Ratio énergie/poids en kcal/kg" pour simplifier les opérations
# Effaçage de toutes les valeurs nulles de Ratio

Tableau5_Ratio = Tableau4.rename(columns = {"Ratio énergie/poids en kcal/kg":"A"})
Tableau5_Ratio = Tableau5_Ratio.drop(Tableau5_Ratio[Tableau5_Ratio.A == 0].index)

# Suppression de la colonne "Pourcentage de protéines"

Tableau5_Ratio = Tableau5_Ratio.drop(columns=["Pourcentage de protéines"])

# Calcul de la moyenne de chaque ratio de produit par pays

Tableau5_Ratio = Tableau5_Ratio.groupby("Produit").mean()

# Renommage de la colonne "A"

Tableau5_Ratio = Tableau5_Ratio.rename(columns = {"A" : "Moyenne du ratio énergie/poids sur tous les pays"})

Tableau5_Ratio.sort_values(by="Moyenne du ratio énergie/poids sur tous les pays", ascending=False).head(20)

Moyenne du ratio énergie/poids sur tous les pays
Produit                                                                   
Huiles de Foie de Poisso                                      11809.003531
Huile de Sésame                                                9559.523810
Huile de Germe de Maïs                                         8950.891229
Huiles de Poissons                                             8917.431319
Huile d'Olive                                                  8774.038462
Huile de Tournesol                                             8769.634357
Huil Plantes Oleif Autr                                        8754.156519
Huile de Soja                                                  8737.921301
Huile de Palme                                                 8682.340531
Huile de Colza&Moutarde                                        8548.729483
Huile Graines de Coton                                         8478.967032
Huile de Palmistes                                             8350.682540
Huile de Coco                                                  8288.877006
Huile de Son de Riz                                            8264.150943
Huile d'Arachide                                               8237.818095
Graisses Animales Crue                                         7562.751875
Beurre, Ghee                                                   7507.566262
Sésame                                                         6136.984310
Arachides Decortiquees                                         5705.476348
Palmistes                                                      5475.000000

<a id="reponse5_2"></a>
#### Liste de 5 aliments parmi les 20 les plus caloriques :
- Les huiles de foie de poisson
- L'huile de sésame
- Les graisses animales crues
- Les arachides décortiquées
- Les palmistes

<a id="reponse5_3"></a>
#### Liste des 20 aliments les plus riches selon les pourcentages moyens :

In [23]:
# Renommage de la colonne "Pourcentage de protéines" pour simplifier les opérations
# Effaçage de toutes les valeurs nulles de pourcentage

Tableau5_Pourcentage = Tableau4.rename(columns = {"Pourcentage de protéines":"A"})
Tableau5_Pourcentage = Tableau5_Pourcentage.drop(Tableau5_Pourcentage[Tableau5_Pourcentage.A == 0].index)

# Suppression de la colonne "Ratio énergie/poids en kcal/kg"

Tableau5_Pourcentage = Tableau5_Pourcentage.drop(columns=["Ratio énergie/poids en kcal/kg"])

# Calcul de la moyenne de chaque pourcentage de produit par pays

Tableau5_Pourcentage = Tableau5_Pourcentage.groupby("Produit").mean()

# Renommage de la colonne "A"

Tableau5_Pourcentage = Tableau5_Pourcentage.rename(columns = {"A" : "Moyenne du pourcentage de protéines des produits sur tous les pays"})

Tableau5_Pourcentage.sort_values(by="Moyenne du pourcentage de protéines des produits sur tous les pays",
                                 ascending=False).head(20)

Moyenne du pourcentage de protéines des produits sur tous les pays
Produit                                                                                     
Graines Colza/Moutarde                                            36.500000                 
Soja                                                              27.254546                 
Arachides Decortiquees                                            25.091310                 
Pois                                                              22.423943                 
Légumineuses Autres                                               22.304684                 
Haricots                                                          21.865805                 
Plantes Oleiferes, Autre                                          18.780182                 
Sésame                                                            18.710715                 
Viande, Autre                                                     17.848150                 
Abats Comestible                                                  17.630901                 
Aliments pour enfants                                             15.431743                 
Cephalopodes                                                      15.035234                 
Viande de Bovins                                                  14.461108                 
Viande d'Ovins/Caprins                                            13.896923                 
Viande de Volailles                                               12.620016                 
Piments                                                           12.431027                 
Poissons Pelagiques                                               12.272671                 
Graines de tournesol                                              12.059930                 
Poissons Marins, Autres                                           11.355373                 
Poivre                                                            11.261152

<a id="reponse5_4"></a>
#### Liste de 5 aliments parmi les 20 les plus riches :
- Les graines colza/moutarde
- Le soja
- Les haricots
- La viande de bovins
- Le poivre

<a id="question6"></a>
#### <span style="background-color: #FFFF00">Question 6 : Calculer la disponibilité intérieure mondiale en kcal pour les végétaux.</span>

In [24]:
# Création du tableau 6

Tableau6 = vegetaux.pivot_table(index =['Zone','Code zone','Produit'],
                                columns='Élément', values='Valeur', aggfunc = sum)
Tableau6 = Tableau6.loc[ : ,('Disponibilité intérieure')]
Tableau6 = Tableau6.reset_index()

# Suppression des valeurs incohérentes

Tableau6 = Tableau6.drop(Tableau6[Tableau6['Disponibilité intérieure']<0].index)
Tableau6 = Tableau6.drop_duplicates()

# Jointure Tableau4 et Tableau6

Tableau6 = pd.merge(Tableau4, Tableau6, how='inner',left_on=['Zone','Produit'],right_on=['Zone','Produit'])

# Suppression de la colonne "Pourcentage de protéines"

DispIntVegeKcal = Tableau6.drop(columns=["Pourcentage de protéines"])

# Renommage des colonnes "Ratio énergie/poids en kcal/kg" et "Disponibilité intérieure en milliers de tonnes"

DispIntVegeKcal = DispIntVegeKcal.rename(columns={"Ratio énergie/poids en kcal/kg" : "A", 
                                                               "Disponibilité intérieure" : "B"})

# Le ratio étant en kcal/kg et la disponibilité intérieure en milliers de tonnes
# il faut multiplier par 1 000 000

DispIntVegeKcal["Disponibilité intérieure en kcal"] = DispIntVegeKcal.A*DispIntVegeKcal.B*1000000

# Suppression des colonnes "A" et "B"

DispIntVegeKcal = DispIntVegeKcal.drop(columns=["A", "B"])

print("La disponibilité intérieure mondiale de végétaux est de",
      DispIntVegeKcal["Disponibilité intérieure en kcal"].sum(),"kcal.")

La disponibilité intérieure mondiale de végétaux est de 1.080098010554526e+16 kcal.


<a id="question7"></a>
#### <span style="background-color: #FFFF00">Question 7 : Combien d'humains pourraient être nourris si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture ? Donner les résultats en terme de calories puis de protéines, et exprimer ces deux résultats en pourcentage de la population mondiale.</span>

In [25]:
# Poids moyen d'un humain : 62 kgs
# Besoin énergétique en kcal : 2500 kcal/jour
# Besoin énergétique en quantité de protéine : 0,9 g/kg/jour

PoidsHumain = 62
BesoinHumainKcalParJour = 2500
BesoinHumainKgProtParJourParKg = 0.9


# Calcul du nombre d'humains nourris

DispoIntMondialeVegetauxKcal = DispIntVegeKcal["Disponibilité intérieure en kcal"].sum()
PopulationMondiale = Pop["Population en milliers d'habitants"].sum()*1000

HumainsNourrisKcal = DispoIntMondialeVegetauxKcal/(BesoinHumainKcalParJour*365)


# Calcul du pourcentage d'humains nourris:

PourcentHumainsNourrisKcal = (HumainsNourrisKcal/PopulationMondiale)*100


# Calcul du pourcentage de protéines de la disponibilité intérieure mondiale

Tableau6["Quantité de protéines de la disponibilité intérieure de végétaux"] = (Tableau6['Pourcentage de protéines']/100)*Tableau6['Disponibilité intérieure']*1000000

HumainsNourrisKgProt = Tableau6["Quantité de protéines de la disponibilité intérieure de végétaux"].sum()/((BesoinHumainKgProtParJourParKg/1000)*PoidsHumain*365)


# Calcul du pourcentage d'humains nourris:

PourcentHumainsNourrisKgProt = (HumainsNourrisKgProt/PopulationMondiale)*100

print("Si toute la disponibilité intérieure mondiale de végétaux était utilisée pour de la nourriture,",
      int(HumainsNourrisKcal), "personnes seraient nourries.")
print("Si toute la disponibilité intérieure mondiale de végétaux était utilisée pour de la nourriture,",
      round(PourcentHumainsNourrisKcal,2),"% de la population mondiale serait nourrie.")
print("Si toute la disponibilité intérieure mondiale de végétaux était utilisée pour de la nourriture,",
      int(HumainsNourrisKgProt), "personnes seraient nourries.")
print("Si toute la disponibilité intérieure mondiale de végétaux était utilisée pour de la nourriture,",
      round(PourcentHumainsNourrisKgProt,2),"% de la population mondiale serait nourrie.")

Si toute la disponibilité intérieure mondiale de végétaux était utilisée pour de la nourriture, 11836690526 personnes seraient nourries.
Si toute la disponibilité intérieure mondiale de végétaux était utilisée pour de la nourriture, 169.16 % de la population mondiale serait nourrie.
Si toute la disponibilité intérieure mondiale de végétaux était utilisée pour de la nourriture, 14369261783 personnes seraient nourries.
Si toute la disponibilité intérieure mondiale de végétaux était utilisée pour de la nourriture, 205.35 % de la population mondiale serait nourrie.


<a id="question8"></a>
#### <span style="background-color: #FFFF00">Question 8 : Combien d'humains pourraient être nourris si toute la disponibilité alimentaire de produits végétaux, la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisées pour de la nourriture ? Donner les résultats en terme de calories puis de protéines, et exprimer ces deux résultats en pourcentage de la population mondiale.</span>

In [26]:
# Création du tableau 8

Tableau8 = vegetaux.pivot_table(index =['Zone','Code zone','Produit'],columns='Élément', values='Valeur', aggfunc = sum)
Tableau8 = Tableau8.loc[ : ,('Disponibilité alimentaire en quantité (kg/personne/an)','Aliments pour animaux','Pertes')]
Tableau8 = Tableau8.reset_index()

# Suppression des valeurs incohérentes

Tableau8 = Tableau8.drop(Tableau8[Tableau8['Disponibilité alimentaire en quantité (kg/personne/an)']<0].index)
Tableau8 = Tableau8.drop(Tableau8[Tableau8['Aliments pour animaux']<0].index)
Tableau8 = Tableau8.drop(Tableau8[Tableau8['Pertes']<0].index)
Tableau8 = Tableau8.drop_duplicates()
Tableau8 = Tableau8.fillna(0)

# Jointure avec Pop

Tableau8 = pd.merge(Pop,Tableau8)
Tableau8['Disponibilité alimentaire en quantité (kg/personne/an)'] = Tableau8['Disponibilité alimentaire en quantité (kg/personne/an)']*Tableau8["Population en milliers d'habitants"]*1000/1000000

#Calcul Disponibilité totale en milliers de tonnes

Tableau8['Disponibilité alimentaire en quantité+semences+pertes'] = Tableau8['Disponibilité alimentaire en quantité (kg/personne/an)']+Tableau8['Aliments pour animaux']+Tableau8['Pertes']

# Jointure avec Pop

Tableau8 = pd.merge(Tableau8,Tableau4)

# Conversion en kcal

Tableau8['Disponibilité alimentaire en quantité+semences+pertes en kcal'] = Tableau8['Disponibilité alimentaire en quantité+semences+pertes']*Tableau8['Ratio énergie/poids en kcal/kg']*1000000

# Conversion en kg de protéines

Tableau8['Disponibilité alimentaire en quantité+semences+pertes en kg de protéines'] = Tableau8['Disponibilité alimentaire en quantité+semences+pertes']*(Tableau8['Pourcentage de protéines']/100)*1000000

DispoSemencesPertesKcal = Tableau8['Disponibilité alimentaire en quantité+semences+pertes en kcal'].sum()
DispoSemencesPertesKgProt = Tableau8['Disponibilité alimentaire en quantité+semences+pertes en kg de protéines'].sum()

HumainsNourrisKcal_8 = DispoSemencesPertesKcal/(BesoinHumainKcalParJour*365)
PourcentHumainsNourrisKcal_8 = (HumainsNourrisKcal_8/PopulationMondiale)*100
HumainsNourrisKgProt_8 = DispoSemencesPertesKgProt/((BesoinHumainKgProtParJourParKg/1000)*PoidsHumain*365)
PourcentHumainsNourrisKgProt_8 = (HumainsNourrisKgProt_8/PopulationMondiale)*100

print("Si toute la disponibilité alimentaire en quantité, les semences et les pertes mondiale de végétaux étaient utilisées pour de la nourriture :")
print("En terme de besoin en kcal :")
print("-",int(HumainsNourrisKcal_8), "d'êtres humains seraient nourris.")
print("- soit",round(PourcentHumainsNourrisKcal_8,2),"% de la population mondiale.")
print("En terme de besoin en protéines :")
print("-", int(HumainsNourrisKgProt_8), "d'êtres humains seraient nourris.")
print("- soit",round(PourcentHumainsNourrisKgProt_8,2),"% de la population mondiale serait nourrie.")

Si toute la disponibilité alimentaire en quantité, les semences et les pertes mondiale de végétaux étaient utilisées pour de la nourriture :
En terme de besoin en kcal :
- 9016929850 d'êtres humains seraient nourris.
- soit 128.86 % de la population mondiale.
En terme de besoin en protéines :
- 10035819925 d'êtres humains seraient nourris.
- soit 143.42 % de la population mondiale serait nourrie.


<a id="question9"></a>
#### <span style="background-color: #FFFF00">Question 9 : Combien d'humains pourraient être nourris avec la disponibilité alimentaire mondiale ? Donner les résultats en terme de calories puis de protéines, et exprimer ces deux résultats en pourcentage de la population mondiale.</span>

In [27]:
# Les disponibilités mondiales en kcal et en kg de protéines ont été calculés question 3

DispoAlimMondialeKcal = Tableau3['Disponibilité alimentaire (Kcal/an)'].sum()
DispoAlimMondialeKgProt = Tableau3['Disponibilité de protéines en quantité (kg/an)'].sum()

HumainsNourrisKcal_9 = DispoAlimMondialeKcal/(BesoinHumainKcalParJour*365)
PourcentHumainsNourrisKcal_9 = (HumainsNourrisKcal_9/PopulationMondiale)*100
HumainsNourrisKgProt_9 = DispoAlimMondialeKgProt/((BesoinHumainKgProtParJourParKg/1000)*PoidsHumain*365)
PourcentHumainsNourrisKgProt_9 = (HumainsNourrisKgProt_9/PopulationMondiale)*100

print("Si toute la disponibilité alimentaire mondiale était utilisée pour de la nourriture :")
print("En terme de besoin en kcal :")
print("-",int(HumainsNourrisKcal_9), "d'êtres humains seraient nourris.")
print("- soit",round(PourcentHumainsNourrisKcal_9,2),"% de la population mondiale.")
print("En terme de besoin en protéines :")
print("-", int(HumainsNourrisKgProt_9), "d'êtres humains seraient nourris.")
print("- soit",round(PourcentHumainsNourrisKgProt_9,2),"% de la population mondiale serait nourrie.")

Si toute la disponibilité alimentaire mondiale était utilisée pour de la nourriture :
En terme de besoin en kcal :
- 8071077237 d'êtres humains seraient nourris.
- soit 115.35 % de la population mondiale.
En terme de besoin en protéines :
- 10177068339 d'êtres humains seraient nourris.
- soit 145.44 % de la population mondiale serait nourrie.


<a id="question10"></a>
#### <span style="background-color: #FFFF00">Question 10 : Quelle est la proportion de la population mondiale considérée comme étant en sous-nutrition ?</span>

In [28]:
sousalimentation.head(3)

Code Domaine                             Domaine  Code zone         Zone  \
0           FS  Données de la sécurité alimentaire          2  Afghanistan   
1           FS  Données de la sécurité alimentaire          2  Afghanistan   
2           FS  Données de la sécurité alimentaire          2  Afghanistan   

   Code Élément Élément  Code Produit  \
0          6132  Valeur        210011   
1          6132  Valeur        210011   
2          6132  Valeur        210011   

                                             Produit  Code année      Année  \
0  Nombre de personnes sous-alimentées (millions)...    20122014  2012-2014   
1  Nombre de personnes sous-alimentées (millions)...    20132015  2013-2015   
2  Nombre de personnes sous-alimentées (millions)...    20142016  2014-2016   

      Unité Valeur Symbole Description du Symbole  Note  
0  millions    7.9       F         Estimation FAO   NaN  
1  millions    8.8       F         Estimation FAO   NaN  
2  millions    9.6       F         Estimation FAO   NaN

In [29]:
Tableau10 = sousalimentation.loc[ : , ("Zone", "Année", "Valeur")]
Tableau10 = Tableau10[Tableau10.Année== "2012-2014"]

# Suppression des valeurs NaN

Tableau10 = Tableau10.drop_duplicates()
Tableau10 = Tableau10.dropna()

# Conversion des données de la colonne "Valeur" en données numériques 

PopSousAlimentée = pd.to_numeric(Tableau10.Valeur, errors='coerce').sum()*1000000

print("En 2013," ,round((PopSousAlimentée/PopulationMondiale)*100,2) ,"% de la poulation mondiale est sous-alimentée.")

En 2013, 10.63 % de la poulation mondiale est sous-alimentée.


<a id="question11"></a>
#### <span style="background-color: #FFFF00">Question 11 :  En ne prenant en compte que les céréales destinées à l'alimentation (humaine et animale), quelle proportion (en terme de poids) est destinée à l'alimentation animale ? Préalablement, établir la liste de produit (ainsi que de leur code) considérés comme des céréales selon la FAO. Repérer les informations contenant les céréales (par exemple en créant une colonne de type booléen nommées "is_cereal").</span>

In [30]:
listeCereales = list(cereales.Produit.unique())

vegetaux["is_cereal"]= vegetaux.Produit.isin(listeCereales)

Tableau11 = vegetaux.loc[ : ,('Zone', 'Élément', 'Code Produit', 'Produit', 'Valeur', 'is_cereal')]
Tableau11 = Tableau11[Tableau11.is_cereal == True]
Tableau11 = Tableau11.drop_duplicates()
Tableau11 = Tableau11.drop(Tableau11[Tableau11.Valeur<0].index)

vegetauxNourriture = Tableau11[Tableau11.Élément=="Nourriture"]
vegetauxAnimaux = Tableau11[Tableau11.Élément=="Aliments pour animaux"]

PropNourritureAnimaux = (vegetauxAnimaux.Valeur.sum()/(vegetauxAnimaux.Valeur.sum()+vegetauxNourriture.Valeur.sum()))*100

print("En ne prenant en compte que les céréales destinées à l'alimentation (humaine et animale),", 
      round(PropNourritureAnimaux),
      "% de leur poids total est destiné à l'alimentation animale.")

En ne prenant en compte que les céréales destinées à l'alimentation (humaine et animale), 46.0 % de leur poids total est destiné à l'alimentation animale.


<a id="question12"></a>
#### <span style="background-color: #FFFF00">Question 12 :</span> 

<a id="réponse12_1"></a>
#### Informations, parmi les données des bilans alimentaires, relatives aux pays dans lesquels la FAO recence des personnes en sous-nutrition :

In [31]:
InfoPaysSousALim = Tableau10.loc[ : , ('Zone')]
InfoPaysSousALim = pd.merge(InfoPaysSousALim, data)
InfoPaysSousALim = InfoPaysSousALim.loc[ : , ('Zone', 'Élément', 'Produit', 'Unité', 'Valeur')]
InfoPaysSousALim = InfoPaysSousALim.drop_duplicates()

InfoPaysSousALim.head(3)

Zone                   Élément           Produit  \
0  Afghanistan                Production  Viande de Bovins   
1  Afghanistan   Importations - Quantité  Viande de Bovins   
2  Afghanistan  Disponibilité intérieure  Viande de Bovins   

                Unité  Valeur  
0  Milliers de tonnes   134.0  
1  Milliers de tonnes     6.0  
2  Milliers de tonnes   140.0

<a id="réponse12_2"></a>
#### Les 15 produits les plus exportés par les pays recensant des personnes en sous-nutrition :

In [32]:
Top15Exportations = InfoPaysSousALim[InfoPaysSousALim.Élément == "Exportations - Quantité"]
Top15Exportations = Top15Exportations.groupby(['Zone','Produit'], as_index=False).sum()
Top15Exportations = Top15Exportations.sort_values(by='Valeur', ascending=False)
Top15Exportations = Top15Exportations.drop_duplicates(subset="Produit")
Top15Exportations = Top15Exportations.head(15)

Top15Exportations

Zone              Produit   Valeur
6779            Thaïlande               Manioc  25214.0
3266            Indonésie       Huile de Palme  23333.0
446             Argentine                 Maïs  20241.0
3209                 Inde     Riz (Eq Blanchi)  11337.0
1674  Chine, continentale      Légumes, Autres   8084.0
472             Argentine                 Soja   7783.0
3143                 Inde                  Blé   7168.0
6810            Thaïlande        Sucre Eq Brut   6462.0
7711             Équateur              Bananes   5956.0
5809                Pérou  Poissons Pelagiques   5725.0
438             Argentine        Huile de Soja   4378.0
455             Argentine                 Orge   4112.0
1712  Chine, continentale              Tomates   3977.0
1699  Chine, continentale               Pommes   3854.0
442             Argentine   Lait - Excl Beurre   2782.0

<a id="réponse12_3"></a>
#### Classement des 200 plus grandes importations de ces 15 produits :

In [33]:
ImpMondiale = data[data.Élément == "Importations - Quantité"]
ImpMondiale = ImpMondiale.loc[ : ,('Zone','Élément','Produit', 'Valeur')]

ProduitsTop15Exportations = Top15Exportations.loc[ : , ('Produit')]

ImpTop15Exp = pd.merge(ImpMondiale,ProduitsTop15Exportations)

Top200Imp = ImpTop15Exp.sort_values(by='Valeur', ascending=False)
Top200Imp = Top200Imp.head(200)

Top200Imp

Zone                  Élément  \
2409             Chine, continentale  Importations - Quantité   
2255             Chine, continentale  Importations - Quantité   
778                            Japon  Importations - Quantité   
529                  Arabie saoudite  Importations - Quantité   
221                           Égypte  Importations - Quantité   
...                              ...                      ...   
185                         Autriche  Importations - Quantité   
206                            Chili  Importations - Quantité   
1289  Iran (République islamique d')  Importations - Quantité   
132              République de Corée  Importations - Quantité   
1327                         Nigéria  Importations - Quantité   

                 Produit   Valeur  
2409                Soja  63381.0  
2255              Manioc  29046.0  
778                 Maïs  14403.0  
529                 Orge  10547.0  
221                  Blé  10331.0  
...                  ...      ...  
185                  Blé   1031.0  
206                  Blé   1022.0  
1289      Huile de Palme   1021.0  
132   Lait - Excl Beurre   1014.0  
1327      Huile de Palme    999.0  

[200 rows x 4 columns]

<a id="réponse12_4"></a>
#### Somme des 200 importations les plus importantes des 15 produits les plus exportés :

In [34]:
Gr_Top200Imp = Top200Imp.groupby(['Produit']).sum()

ProduitsTop200Imp = Gr_Top200Imp.rename(columns = {"Valeur":"Poids total des 200 importations en milliers de tonnes"})

ProduitsTop200Imp

Poids total des 200 importations en milliers de tonnes
Produit                                                                    
Bananes                                                         9663.0     
Blé                                                           149629.0     
Huile de Palme                                                 31767.0     
Huile de Soja                                                   2266.0     
Lait - Excl Beurre                                             90464.0     
Légumes, Autres                                                28177.0     
Manioc                                                         34655.0     
Maïs                                                          102451.0     
Orge                                                           22387.0     
Poissons Pelagiques                                            13445.0     
Pommes                                                          8518.0     
Riz (Eq Blanchi)                                               13990.0     
Soja                                                           93455.0     
Sucre Eq Brut                                                  34208.0     
Tomates                                                         7143.0

<a id="réponse12_5"></a>
#### Tableau pour chaque produit des ratios : 
- entre quantité destinés aux "Autres utilisations" et la disponibilité intérieure
- entre quantité destinés à la nourriture animale et celle de la nourriture totale (animale + humaine)

In [40]:
# Restriction de "data" n'affichant que les "Autres utilisations (non alimentaire)"

AutresUtilisationsMondiale = data[data.Élément == "Autres utilisations (non alimentaire)"]

# Restriction n'affichant que la disponibilité intérieure des 15 produits les plus exportés
# par les pays recensant des populations sous-alimentées

AutresUtilisationsMondiale15Produits = pd.merge(ProduitsTop15Exportations, AutresUtilisationsMondiale)
AutresUtilisationsMondiale15Produits = AutresUtilisationsMondiale15Produits.drop_duplicates()
AutresUtilisationsMondiale15Produits = AutresUtilisationsMondiale15Produits.loc[ : , ('Produit', 'Zone', 'Valeur')]
AutresUtilisationsMondiale15Produits = AutresUtilisationsMondiale15Produits.groupby(['Produit']).sum()

# Renommage de la colonne "Valeur" pour opération

AutresUtilisationsMondiale15Produits = AutresUtilisationsMondiale15Produits.rename(columns={'Valeur':'A'})

# Restriction de "data" n'affichant que la disponibilité intérieure

DispIntMondiale = data[data.Élément == "Disponibilité intérieure"]

# Restriction n'affichant que la disponibilité intérieure des 15 produits les plus exportés
# par les pays recensant des populations sous-alimentées

DispIntMondiale15Produits = pd.merge(ProduitsTop15Exportations, DispIntMondiale)
DispIntMondiale15Produits = DispIntMondiale15Produits.drop_duplicates()
DispIntMondiale15Produits = DispIntMondiale15Produits.drop(DispIntMondiale15Produits[DispIntMondiale15Produits.Valeur<0].index)
DispIntMondiale15Produits = DispIntMondiale15Produits.loc[ : , ('Produit', 'Zone', 'Valeur')]
DispIntMondiale15Produits = DispIntMondiale15Produits.groupby(['Produit']).sum()

# Renommage de la colonne "Valeur" pour opération

DispIntMondiale15Produits = DispIntMondiale15Produits.rename(columns={'Valeur':'B'})

# Restriction de "data" n'affichant que les "Aliments pour animaux"

AlimentsAnimauxMondiale = data[data.Élément == "Aliments pour animaux"]

# Restriction n'affichant que la disponibilité intérieure des 15 produits les plus exportés
# par les pays recensant des populations sous-alimentées

AlimentsAnimauxMondiale15Produits = pd.merge(ProduitsTop15Exportations, AlimentsAnimauxMondiale)
AlimentsAnimauxMondiale15Produits = AlimentsAnimauxMondiale15Produits.drop_duplicates()
AlimentsAnimauxMondiale15Produits = AlimentsAnimauxMondiale15Produits.loc[ : , ('Produit', 'Zone', 'Valeur')]
AlimentsAnimauxMondiale15Produits = AlimentsAnimauxMondiale15Produits.groupby(['Produit']).sum()

# Renommage de la colonne 'Valeur' pour opération

AlimentsAnimauxMondiale15Produits = AlimentsAnimauxMondiale15Produits.rename(columns={'Valeur':'C'})

# Restriction de "data" n'affichant que les éléments "Nourriture"

NourritureMondiale = data[data.Élément == "Nourriture"]

# Restriction n'affichant que la disponibilité intérieure des 15 produits les plus exportés
# par les pays recensant des populations sous-alimentées

NourritureMondiale15Produits = pd.merge(ProduitsTop15Exportations, NourritureMondiale)
NourritureMondiale15Produits = NourritureMondiale15Produits.drop_duplicates()
NourritureMondiale15Produits = NourritureMondiale15Produits.loc[ : , ('Produit', 'Zone', 'Valeur')]
NourritureMondiale15Produits = NourritureMondiale15Produits.groupby(['Produit']).sum()

# Renommage de la colonne "Valeur" pour opération

NourritureMondiale15Produits = NourritureMondiale15Produits.rename(columns={'Valeur':'D'})

# Jointure de "AutresUtilisationsMondiale15Produits" et"DispIntMondiale15Produits"

Jointure1 = pd.merge(AutresUtilisationsMondiale15Produits, DispIntMondiale15Produits, on='Produit')
Jointure2 = pd.merge(AlimentsAnimauxMondiale15Produits, NourritureMondiale15Produits, on='Produit')

Tableau12 = pd.merge(Jointure1,Jointure2, on='Produit')
Tableau12['Ratio  "Autres utilisations" / Disponibilité intérieure'] = Tableau12.A/Tableau12.B
Tableau12['Ratio  Aliments pour animaux / Nourriture totale (animale + humaine)'] = Tableau12.C/(Tableau12.C+Tableau12.D)
Tableau12 = Tableau12.drop(columns=['A','B','C','D'])

Tableau12.head(3)

Ratio  "Autres utilisations" / Disponibilité intérieure  \
Produit                                                                   
Bananes                                                  0.001365         
Blé                                                      0.033001         
Huile de Palme                                           0.698109         

                Ratio  Aliments pour animaux / Nourriture totale (animale + humaine)  
Produit                                                                               
Bananes                                                  0.010032                     
Blé                                                      0.220714                     
Huile de Palme                                           0.000000

<a id="réponse12_6"></a>
#### Affichage des trois produits avec le ratio "Autres utilisations" / Disponibilité intérieure le plus élevé :

In [36]:
Top3MaxRatioAutreUtiDispInt = Tableau12.loc[ : ,('Ratio  "Autres utilisations" / Disponibilité intérieure')]
Top3MaxRatioAutreUtiDispInt = Top3MaxRatioAutreUtiDispInt.reset_index()
Top3MaxRatioAutreUtiDispInt = Top3MaxRatioAutreUtiDispInt.sort_values(by='Ratio  "Autres utilisations" / Disponibilité intérieure',ascending=False).head(3)

Top3MaxRatioAutreUtiDispInt

Produit  Ratio  "Autres utilisations" / Disponibilité intérieure
2  Huile de Palme                                           0.698109      
3   Huile de Soja                                           0.416172      
7            Maïs                                           0.198144

<a id="réponse12_7"></a>
#### Affichage des trois produits avec le ratio Aliments pour animaux / Nourriture totale (animale + humaine) le plus élevé :

In [37]:
Top3MaxRatioAlimAnimNourrTotale = Tableau12.loc[ : ,('Ratio  Aliments pour animaux / Nourriture totale (animale + humaine)')]
Top3MaxRatioAlimAnimNourrTotale = Top3MaxRatioAlimAnimNourrTotale.reset_index()
Top3MaxRatioAlimAnimNourrTotale = Top3MaxRatioAlimAnimNourrTotale.sort_values(by='Ratio  Aliments pour animaux / Nourriture totale (animale + humaine)',ascending=False).head(3)

Top3MaxRatioAlimAnimNourrTotale

Produit  \
8     Orge   
7     Maïs   
12    Soja   

    Ratio  Aliments pour animaux / Nourriture totale (animale + humaine)  
8                                            0.931686                     
7                                            0.813520                     
12                                           0.621391

<a id="question13"></a>
#### <span style="background-color: #FFFF00">Question 13 : Combien de tonnes de céréales pourraient être libérées si les USA diminuaient leur production de produits animaux de 10% ?</span>

In [38]:
vegetauxAlimAnimaux = vegetaux[vegetaux.Élément=="Aliments pour animaux"]

vegetauxAlimAnimauxUSA = vegetauxAlimAnimaux[vegetauxAlimAnimaux.Zone== "États-Unis d'Amérique"]

cerealesZoneProduit = cereales.loc[ : ,('Zone', 'Produit')]

AlimAnimauxUSACereales = pd.merge(vegetauxAlimAnimauxUSA,cerealesZoneProduit)

TonnesCerealesLiberees = AlimAnimauxUSACereales.Valeur.sum()*0.1

print("Si les USA diminuaient leur production de produits animaux de 10%,", TonnesCerealesLiberees, "milliers de tonnes seraient libérées.")

Si les USA diminuaient leur production de produits animaux de 10%, 14009.6 milliers de tonnes seraient libérées.


<a id="question14"></a>
#### <span style="background-color: #FFFF00">Question 14 : En Thaïlande, quelle proportion de manioc est exportée ? Quelle est la proportion de personnes en sous-nutrition ?</span>

In [39]:
# Calcul de la proportion de manioc

vegetauxThaïlande = vegetaux[vegetaux.Zone=="Thaïlande"]

ManiocThaïlande = vegetauxThaïlande[vegetauxThaïlande.Produit=="Manioc"]

ProdManiocThaïlande = ManiocThaïlande[ManiocThaïlande.Élément == "Production"]

ExpManiocThaïlande = ManiocThaïlande[ManiocThaïlande.Élément == "Exportations - Quantité"]

PropExpManiocThaïlande = (ExpManiocThaïlande.Valeur.sum()/ProdManiocThaïlande.Valeur.sum())*100

# Calcul de la proportion de personnes en sous-nutrition

sousalimentationThaïlande2013 = sousalimentation[sousalimentation.Zone == "Thaïlande"]
sousalimentationThaïlande2013 = sousalimentationThaïlande2013[sousalimentationThaïlande2013.Année == "2012-2014"]
sousalimentationThaïlande2013 = pd.to_numeric(sousalimentationThaïlande2013.Valeur, errors='coerce')

PopTotaleThaïlande = Pop[Pop.Zone == "Thaïlande"]
PopTotaleThaïlande = PopTotaleThaïlande["Population en milliers d'habitants"].sum()

PropPersSousAlimThaïlande2013 = ((sousalimentationThaïlande2013.sum()*1000)/PopTotaleThaïlande)*100

# Résultats

print("En 2013 en Thaïlande :")
print("-",round(PropExpManiocThaïlande,2), "% de la production de manioc est exportée.")
print("- la proportion de personnes sous-alimentée est de",round(PropPersSousAlimThaïlande2013,2),"%.")

En 2013 en Thaïlande :
- 83.41 % de la production de manioc est exportée.
- la proportion de personnes sous-alimentée est de 8.36 %.
